packages/tupaia-web/src/features/EntitySearch/EntitySearch.tsx

```diff
@@ -63,7 +63,8 @@ export const EntitySearch = () => {
   };
 
   const toggleExpandedSearchBar = () => {
-    setIsOpen(!isOpen);
+    const updatedOpenValue = !isOpen;
+    setIsOpen(updatedOpenValue);
     gaEvent('Search', 'Toggle Expand');
     if (isOpen === false) {
```

Comment:
Just need to use `updatedOpenValue === false` in the check instead of `isOpen === false`